tests/jax/systems/mappo_system_test.py

```diff
@@ -54,36 +78,37 @@ def test_except_trainer(
 
     # Optimizer.
     optimizer = optax.chain(
-        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+        optax.clip_by_global_norm(40.0),
+        optax.adam(1e-4),
     )
 
     # Build the system
     test_full_system.build(
         environment_factory=environment_factory,
         network_factory=network_factory,
         logger_factory=logger_factory,
-        checkpoint_subpath=checkpoint_subpath,
+        experiment_path=checkpoint_subpath,
         optimizer=optimizer,
-        executor_parameter_update_period=20,
-        multi_process=False,
+        executor_parameter_update_period=1,
+        multi_process=True,
         run_evaluator=True,
         num_executors=1,
+        max_queue_size=500,
         use_next_extras=False,
-        sample_batch_size=2,
+        sample_batch_size=5,
+        checkpoint=True,
+        nodes_on_gpu=[],
+        lp_launch_type=lp.LaunchType.TEST_MULTI_THREADING,
     )
 
-    (
-        data_server,
-        parameter_server,
-        executor,
-        evaluator,
-        trainer,
-    ) = test_full_system._builder.store.system_build
-
-    assert isinstance(executor, acme.core.Worker)
+    (trainer_node,) = test_full_system._builder.store.program._program._groups[
+        "trainer"
+    ]
+    trainer_node.disable_run()
+    test_address_builder.bind_addresses([trainer_node])
 
-    # Step the executor
-    executor.run_episode()
+    test_full_system.launch()
+    trainer_run = trainer_node.create_handle().dereference()
 
-    # Step the trainer
-    trainer.step()
+    for _ in range(5):
+        trainer_run.step()
```

Comment:
How is the trainer stepping if we haven't run the executor yet? Or is the executor running in its own thread when launch is called?

Reply:
Okay about that I talked with Ruan about it the step here is not the step function from the trainer it is an internal function in the node to run the process

Reply:
the executor is running while we add system.launch

Reply:
Is the trainer not then also running when we call system.launch?

Reply:
Happy to merge :)